### DISCUSSION TITLE
Create mesh using a stream of STL file given via stdin

### DISCUSSION THREAD

**Discussion by AlbertDeTerre** [CGAL/cgal#8144] [Q&A]
Hello, I would like to create a Surface_mesh object using a stream of an STL file that I receive via STDIN.
When I try to create the mesh using the stream received, the mesh is not created... Why does this happen?

## Code

```cpp
#include <iostream>

#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
#include <CGAL/Surface_mesh.h>

using K = CGAL::Exact_predicates_inexact_constructions_kernel;
using Surface_mesh = CGAL::Surface_mesh<K::Point_3>;

int main(int argc, char **argv)
{
    // Retrieve the stream of STL file
    std::istream &input_stream = std::cin;

    Surface_mesh mesh;

    if (!CGAL::IO::read_STL(input_stream, mesh)){
        cout << "Can't create the mesh" << endl;
    };

     return 0;
}
```

## Ouput

`Can't create the mesh`

## Additional information

- STL file used is a binary STL file
- If I use a filepath to load the STL, it works (so STL file is not corrupted or bad)

**Comment by AlbertDeTerre** [comment]
## Update !

I found that using a stringstream instead of an istream make it work:

```cpp
std::stringstream input_stream;
input_stream << std::cin::rdbuf << endl;
```

Is there a explanation to that ?

**Comment by lrineau** [comment]
> Is there a explanation to that ?

Not really. Have you found clues since then?
